lib/dns/dns.go

```diff
@@ -356,34 +398,31 @@ func (s *Server) HandleVIC(w mdns.ResponseWriter, r *mdns.Msg) (bool, error) {
 	if e == nil {
 		return false, fmt.Errorf("Could not find requesting container with ip %s", ip)
 	}
-	scope := e.Scope()
 
-	if domain != "" && scope.Name() != domain {
-		return false, fmt.Errorf("Inter-scope request for container %s in %s from %s", name, domain, scope.Name())
+	if domain != "" && e.Scope().Name() != domain {
+		return false, fmt.Errorf("Intra-scope request for container %s in %s from %s", name, domain, e.Scope().Name())
 	}
 
-	// container specific alias search
-	cons := ctx.ContainersByAlias(fmt.Sprintf("%s:%s:%s", scope.Name(), e.Container().Name(), name))
-	if len(cons) == 0 {
-		// scope-wide search
-		cons = ctx.ContainersByAlias(fmt.Sprintf("%s:%s", scope.Name(), name))
+	eps := lookupByAlias(ctx, e.Scope(), e.Container(), name)
+	if len(eps) == 0 {
+		// lookup container by name
+		eps = lookupByName(ctx, e.Container(), name)
 	}
 
-	if len(cons) == 0 {
+	if len(eps) == 0 {
 		log.Debugf("Can't find the container: %q", name)
 		return false, fmt.Errorf("Can't find the container: %q", name)
 	}
 
 	// FIXME: Add AAAA when we support it
-	answer := make([]mdns.RR, len(cons))
-	// shuffle cons
-	for i := 0; i < len(cons); i++ {
-		j := random.Intn(len(cons))
-		cons[i], cons[j] = cons[j], cons[i]
+	answer := make([]mdns.RR, len(eps))
+	// shuffle eps
+	for i := 0; i < len(eps); i++ {
+		j := random.Intn(len(eps))
```

Comment:
this shuffle needs to be fixed. I and Hasan talked in chat. Here is a link Hasan found that explains why https://blog.codinghorror.com/the-danger-of-naivete/